packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -221,8 +220,9 @@ export class H5GroveApi extends DataProviderApi {
     // Treat 'other' entities as unresolved
     return {
       ...baseEntity,
+      attributes: [],
       kind: EntityKind.Unresolved,
-    } as UnresolvedEntity;
+    };
```

Comment:
`attributes` were undefined for "other" entities.

It was not caught by TS due to the type assertion `as UnresolvedEntity` that allows missing/additionnal field in an object.

To avoid such issues in the future, we should perhaps use type declaration rather than type assertion. See [this simple example](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGJQK7DMg3gKGWQAs4BnABWAAcyAuZAIwHtmAbCOEAbkOQXbMoDMmCigA5rwC++fKEixEKAILVqHZBAAekEABMyaTNjx8AbnHEQwATxFjJMufhgYQCMMGYh+UTpAAQnD66FhgABTAZGoaEAws7JwgAJQMYaYERAIgokzkEBk4ALx4AmxCDADkEv4QIFUANCTkVLQMYhgQsnzAMFEx6hwpZkREAPTjyACy0WSSyDDAEGz6TBg4ACoAysj6zBBGIMw4AgC2GnCgfET+YBhQvlljRAB074wFRY03yNLI5GQsQ4fB6txsD18nzIhRMYHwPTcHi8Pj8AQgAHFWKE4QNgfEmKwOFw0sZwqN+D48tDYeTSrhypVkDU6g1mqRKDR6MhOt05EQ+nihhARs8JlNNrZqCh9BAEGwrHAUb4znAANaHZA7SkXBWgZB9AGLZarA1GM5zSTjEKy-S-HJ5ODChj45D096vGnfZCWax2aoSdiyhqyF53SEA4W8Ihg5Dhx75GFFBH4IA)

Reply:
Good point, the extra variable is annoying, but if it detects this sort of errors, it is indeed worth it.